--- a/drivers/net/dsa/qca/qca8k-8xxx.c
+++ b/drivers/net/dsa/qca/qca8k-8xxx.c
@@ -2022,6 +2022,12 @@ qca8k_setup(struct dsa_switch *ds)
 			return ret;
 		}
+		/* Disable learning by default on all ports */
 	}
 
 	/* Forward all unknown frames to CPU port for Linux processing */
@@ -2046,11 +2052,6 @@ qca8k_setup(struct dsa_switch *ds)
 		if (ret)
 			return ret;
 
-		ret = regmap_clear_bits(priv->regmap, QCA8K_PORT_LOOKUP_CTRL(port),
-					QCA8K_PORT_LOOKUP_LEARN);
-		if (ret)
-			return ret;
-
 		ret = qca8k_rmw(priv, QCA8K_PORT_LOOKUP_CTRL(cpu_port),
 				BIT(port), BIT(port));
 		if (ret)
@@ -2107,6 +2108,9 @@ qca8k_setup(struct dsa_switch *ds)
 	/* Set max number of LAGs supported */
 	ds->num_lag_ids = QCA8K_NUM_LAGS;
 
+	ds->assisted_learning_on_cpu_port = true;
+
 	return 0;
 }
